File: php/src/Model/QueueAuditSweepBacklog.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * How much work the background sweeps still owe.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class QueueAuditSweepBacklog
{
    public function __construct(
        public readonly int $scheduleActivations,
        public readonly int $scheduleMisses,
        public readonly int $responseDeadlines,
        public readonly int $completionDeadlines,
        public readonly int $slaExpiries,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            scheduleActivations: (int) ($data['scheduleActivations'] ?? 0),
            scheduleMisses: (int) ($data['scheduleMisses'] ?? 0),
            responseDeadlines: (int) ($data['responseDeadlines'] ?? 0),
            completionDeadlines: (int) ($data['completionDeadlines'] ?? 0),
            slaExpiries: (int) ($data['slaExpiries'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/QueueAuditSweepBacklog.java
package io.fivexer.sdk.model;

/**
 * How much work the background sweeps still owe.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class QueueAuditSweepBacklog {
    private Integer scheduleActivations;
    private Integer scheduleMisses;
    private Integer responseDeadlines;
    private Integer completionDeadlines;
    private Integer slaExpiries;

    public Integer getScheduleActivations() { return scheduleActivations; }
    public Integer getScheduleMisses() { return scheduleMisses; }
    public Integer getResponseDeadlines() { return responseDeadlines; }
    public Integer getCompletionDeadlines() { return completionDeadlines; }
    public Integer getSlaExpiries() { return slaExpiries; }
}
